web/src/main/webapp/xslt/common/index-utils.xsl

```diff
@@ -267,8 +267,7 @@
                             select="replace(@locale, '#', '')"/>
               <xsl:variable name="elementLanguage3LetterCode"
                             select="$languages/lang[@id = $elementLanguage]/@value"/>
-
-              <xsl:if test="$elementLanguage3LetterCode != ''">
+              <xsl:if test="$elementLanguage3LetterCode != '' and ($elementLanguage3LetterCode !=$mainLanguage or count($element//(*:CharacterString|*:Anchor)[. != ''])=0) ">
```

Comment:
No. The all point of having one field per language is to be able to apply proper analysis (see https://github.com/geonetwork/core-geonetwork/blob/main/web/src/main/webResources/WEB-INF/data/config/index/records.json#L1072-L1181) and do better search in multilingual env. So here, this could only work if all records are using the same main language which is usually not the case. It is not "indexed twice", it is indexed with the language analyzer in the language field and using the default analyzer in the default one.

Reply:
fact was that with such an md:
https://github.com/geoadmin/geocat/blob/geocat_4.2.x/schemas/iso19139.che/src/test/resources/org/fao/geonet/schemas/call_112.xml

langger is indexed twice:
![twice_langger](https://user-images.githubusercontent.com/18677090/216301603-3bd7db40-824d-4db7-a9d4-59329ea2559e.png)

does this count as an invalid json ?

Reply:
Ah ok I see, Indexing was not reporting any issue but indeed the JSON property for the main language was set twice with the same value.